### ISSUE TITLE
Accept moment objects in ngModel

### ISSUE THREAD

**Issue by oscar09** [mominsamir/smDateTimeRangePicker#101]
see discussion on #93

**Comment by pablorsk** [comment]
Great! Then the proposal is, in date-range-picker case, this:

```javasctipt
ngModel = {
  startDateMoment: moment_object,
  endDateMoment: moment_object
}
```

Another proposal is use this:
```javasctipt
ngModel = {
  startDate: moment_object,
  endDate: moment_object
}
```

Also, callback parameter is not very important now, becouse on callback is possible access to ngModel.

I'm working on a proposal, is not very difficult.

**Comment by oscar09** [comment]
Actually it is already done (the changes on the callback). I will post them in a couple of mins, I am running some tests.

**Comment by oscar09** [comment]
@pablorsk @mominsamir are you able to select previous dates on the demo page for the date range picker? I worked today on this task and noticed this issue. I am not sure if it is related to my forked version that is why I would like to double check with you guys. I pulled the latest changes found on master, and I did a quick debug and noticed that the error comes after commit 981604a7960914be0c046e8f9a09a761ab4ec464 . I still need to debug further to find out where the error was inserted.
